crates/bevy_render/src/render_resource/uniform_vec.rs

```diff
@@ -20,57 +28,68 @@ impl<T: AsStd140> Default for UniformVec<T> {
             values: Vec::new(),
             scratch: Vec::new(),
             uniform_buffer: None,
-            capacity: 0,
-            item_size: (T::std140_size_static() + <T as AsStd140>::Output::ALIGNMENT - 1)
-                & !(<T as AsStd140>::Output::ALIGNMENT - 1),
         }
     }
 }
 
 impl<T: AsStd140> UniformVec<T> {
+    const ITEM_SIZE: usize =
+        (std::mem::size_of::<T::Output>() + <T as AsStd140>::Output::ALIGNMENT - 1)
+            & !(<T as AsStd140>::Output::ALIGNMENT - 1);
+
+    /// Gets the reference to the underlying buffer, if one has been allocated.
     #[inline]
-    pub fn uniform_buffer(&self) -> Option<&Buffer> {
+    pub fn buffer(&self) -> Option<&Buffer> {
         self.uniform_buffer.as_ref()
     }
 
+    /// Creates a binding for the underlying buffer.
+    /// Returns `None` if no buffer has been allocated.
     #[inline]
     pub fn binding(&self) -> Option<BindingResource> {
         Some(BindingResource::Buffer(BufferBinding {
-            buffer: self.uniform_buffer()?,
+            buffer: self.buffer()?,
             offset: 0,
-            size: Some(NonZeroU64::new(self.item_size as u64).unwrap()),
+            size: Some(NonZeroU64::new(Self::ITEM_SIZE as u64).unwrap()),
         }))
     }
 
+    /// Gets the capacity of the underlying buffer.
+    ///
+    /// Will return 0 if no buffer has been allocated yet.
     #[inline]
-    pub fn len(&self) -> usize {
+    pub fn buffer_capacity(&self) -> usize {
         self.values.len()
     }
 
-    #[inline]
-    pub fn is_empty(&self) -> bool {
-        self.values.is_empty()
-    }
-
-    #[inline]
-    pub fn capacity(&self) -> usize {
-        self.capacity
-    }
-
     pub fn push(&mut self, value: T) -> usize {
         let index = self.values.len();
         self.values.push(value);
         index
     }
```

Comment:
Remove? Or you kept it for the index return value? If so, maybe it should also be renamed?
```suggestion
    pub fn push_and_get_offset(&mut self, value: T) -> usize {
        let index = self.values.len();
        self.values.push(value);
        index
    }
```
perhaps?

Reply:
The actual index is used in a few places internally. Should we rename and make it `pub(crate)`?

Reply:
Wait no, it's required by bevy_pbr. I'll just rename it.